lib/graphql/models/definition_helpers/associations.rb

```diff
@@ -41,7 +41,7 @@ def self.resolve_has_one_type(reflection)
           graph_types = valid_types.map { |t| GraphQL::Models.get_graphql_type(t) }.compact
 
           GraphQL::UnionType.define do
-            name "#{model_type.name}#{reflection.foreign_type.classify}"
+            name "#{model_type.class_name}#{reflection.foreign_type.classify}"
```

Comment:
Hmm, I'm not sure if this code works universally. At least on Rails 4, ActiveRecord classes don't respond to `class_name`:
```ruby
class MyModel < ActiveRecord::Base; end
MyModel.class_name
# NoMethodError: undefined method `class_name' for MyModel(Table doesn't exist):Class
# Did you mean?  class_eval
```